### PR TITLE
Hip88

### PR DESCRIPTION
**Pull request by mikev**
Create a silver table consisting of gateway rewards.  This will allow a fast determination of "active" gateways.

Also create a silver table of packets.  Not needed for HIP88, but lets add this now.

### CODE REVIEW

File: manifests/poc-data-cluster/prod/spark/hip88-silver.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+kind: ConfigMap
+metadata:
+  name: hip88-silver-query
+  namespace: spark
+data:
+  query.sql: |
+    WITH raw AS (
+      SELECT
+        distinct gateway
+      FROM packet_router_packet_report_v1
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/hip88-silver.yaml:11]: I think we want a more generally usable packet router packet report silver table. We'll do the select distinct in another query that is unioned with PoC data

**Review by ChewingGlass — CHANGES_REQUESTED** [review]

File: manifests/poc-data-cluster/prod/spark/hip88-silver.yaml (hunk 1 above)
Comment (mikev) [reply manifests/poc-data-cluster/prod/spark/hip88-silver.yaml]: Removed file

File: manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          tolerations: # Schedule executor pods on spot instance group
+            - key: dedicated
+              operator: Equal
+              value: spark
+              effect: NoSchedule
+          nodeSelector:
+            nodegroup-type: spot
+          containers:
+          - name: iot-rewards-delta-lake-sink
+            image: public.ecr.aws/k0m1p4t7/protobuf-delta-lake-sink:0.0.9
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml:24]: Needs to be 0.0.10

File: manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+              - --source-protos
+              - service/poc_lora.proto
+              - --target-bucket
+              - foundation-data-lake-requester-pays
+              - --target-table
+              - bronze/iot_reward_share
+              - --target-region
+              - us-west-2
+              - --partition-timestamp-column
+              - received_timestamp
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml:59]: Timestamp is wrong, and missing the timestamp divisor field. See helium-data scripts.

File: manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+              - --source-proto-name
+              - "iot_reward_share"
+              - --source-proto-base-url
+              - https://raw.githubusercontent.com/helium/proto/master/src
+              - --source-protos
+              - data_rate.proto
+              - --source-protos
+              - region.proto              
+              - --source-protos
+              - service/poc_lora.proto
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml:51]: I don't think this is the right proto. see the helium-data script

**Review by ChewingGlass — CHANGES_REQUESTED** [review]

File: manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+metadata:
+  name: iot-poc-silver-query
+  namespace: spark
+data:
+  query.sql: |
+    SELECT 
+      start_period,
+      end_period,
+      gateway_reward
+      FROM iot_reward_share
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml:12]: If we're only taking gateway_reward, recommend (1) exploding this out by selecting the individual values and (2) filtering where gateway_reward is not null

File: manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+metadata:
+  name: iot-packets-silver-query
+  namespace: spark
+data:
+  query.sql: |
+    SELECT 
+      gateway_tmst,
+      oui,
+      net_id,
+      rssi,
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml:12]: unit

File: manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      frequency,
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml:13]: unit

File: manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+      snr,
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml:14]: unit. And make sure it's matching the other silver table for these units

File: manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+      datarate,
+      region,
+      b58encodeChecked(gateway) as gateway_key,
+      payload_hash,
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml:18]: all bytes should be base64 encoded strings, as in the other silver table

File: manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+            path: query.sql
+  driver:
+    serviceAccount: spark-data-lake-access
+    cores: 1
+    coreLimit: "1200m"
+    memory: "512m"
+    nodeSelector:
+      node.kubernetes.io/instance-type: m5.large
+    envVars:
+      TABLE_LORA_POC_V1: s3a://foundation-data-lake-requester-pays/bronze/packet_router_packet_report_v1
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml:66]: Env var should be `TABLE_PACKET_ROUTER_PACKET_REPORT_V1` if that's how you want it to be named in your select statement

File: manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+            path: query.sql
+  driver:
+    serviceAccount: spark-data-lake-access
+    cores: 1
+    coreLimit: "1200m"
+    memory: "512m"
+    nodeSelector:
+      node.kubernetes.io/instance-type: m5.large
+    envVars:
+      TABLE_LORA_POC_V1: s3a://foundation-data-lake-requester-pays/bronze/lora_poc_v1
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml:52]: fix env var name

File: manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+      PARTITION_BY: "date"
+      CHECKPOINT: s3a://foundation-data-lake-requester-pays/checkpoints/iot-poc-silver
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml:54]: fix checkpoint location

File: manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+      OUTPUT: s3a://foundation-data-lake-requester-pays/silver/iot-poc-witnesses
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml:55]: fix output

File: manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+    volumeMounts:
+      - name: "test-volume"
+        mountPath: "/tmp"
+      - name: config-vol
+        mountPath: /app
+  executor:
+    serviceAccount: spark-data-lake-access
+    cores: 1
+    coreLimit: "1200m"
+    instances: 3
```
Comment (ChewingGlass) [inline manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml:68]: This is a smaller table. Can we get away with fewer instances?

File: manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml (hunk 2 above)
Comment (mikev) [reply manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml]: Fixed

File: manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml (hunk 3 above)
Comment (mikev) [reply manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml]: Fixed

File: manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml (hunk 14 above)
Comment (mikev) [reply manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml]: Reduced to 2

File: manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml (hunk 13 above)
Comment (mikev) [reply manifests/poc-data-cluster/prod/spark/iot-rewards-silver.yaml]: Renamed to TABLE_IOT_PACKETS.
A simpler name is better.  Nobody know what "packet reports v1" means.

File: manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml (hunk 4 above)
Comment (mikev) [reply manifests/poc-data-cluster/prod/helium/iot-rewards-share-delta-lake-sink.yaml:51]: What is wrong here?

File: manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml (hunk 6 above)
Comment (mikev) [reply manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml]: Fixed.

File: manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml (hunk 7 above)
Comment (mikev) [reply manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml]: Fixed.

File: manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml (hunk 8 above)
Comment (mikev) [reply manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml]: Fixed

File: manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml (hunk 9 above)
Comment (mikev) [reply manifests/poc-data-cluster/prod/spark/iot-packets-silver.yaml]: Fixed

**Review by ChewingGlass — APPROVED** [review]

**Review by deasydoesit — APPROVED** [review]
lgtm

**Pull request merged**